src/D3MHub.sol

```diff
@@ -380,6 +382,23 @@ contract D3MHub {
         return ilks[ilk].tic;
     }
 
+    /**
+        @notice Rebalance position if someone permissionlesly paid part of the debt
+        @dev This function will regenerate the debt of the position in case art < ink
+        and send the fresh DAI generated to the surplus buffer.
+        @param ilk bytes32 of the D3M ilk name
+    */
+    function fix(bytes32 ilk) external lock {
```

Comment:
would this require a `vat.live` check (as `suck` is being used)?

Reply:
i kind of think we would still want this right after cage if we are unbalanced.  Perhaps we add a `require(end.debt() == 0)` check to we don't suck after the outstanding dai is accounted for?

Reply:
yeah, after a second look, I might see why it wasn't added but maybe as you pointed out a safeguard could be considered here.

Reply:
A `vat.live` check was introduced in 8eb2048, going to resolve.